kernel/src/process.rs

```diff
@@ -183,21 +183,22 @@ pub fn load_processes<C: Chip>(
                 }
             };
 
-            // Now we can get a slice which only encompasses the app. We will
-            // either parse this as an actual app, or skip over this region.
-            let app_flash = remaining_flash
-                .get(0..app_length as usize)
+            // Now we can get a slice which only encompasses the length of
+            // flash described by this tbf header.  We will either parse this
+            // as an actual app, or skip over this region.
+            let entry_flash = remaining_flash
+                .get(0..entry_length as usize)
                 .ok_or(ProcessLoadError::NotEnoughFlash)?;
 
-            // Check if we should continue parsing this app. If the initial
-            // header parsing failed (i.e. the header_length is 0), we don't
-            // bother.
-            let memory_offset = if header_length > 0 {
+            // Determine how much of the available app memory this entry uses.
+            // Either enough for the newly created process, or none if this
+            // entry is not a process.
+            let process_memory_length = if header_length > 0 {
                 // Try to create a process object from that app slice.
-                let (process, memory_offset) = Process::create(
+                let (process, process_memory_length) = Process::create(
```

Comment:
`process_memory_length` is technically not correct. However, an older PR of mine that I trampled all over with this PR would clarify it, and we should be using slices anyway (another branch I have that I trampled all over with this change). This return value is really `memory_bytes_consumed`, and the process's start of memory does not have to be at the same address of the range passed to `create()`. Just FYI if you are curious, and to help motivate my other (and forthcoming) PRs!

Reply:
Yeah, that all makes sense. I look forward to future PRs and even better names 👍